### ISSUE TITLE
Domains Purge is using deprecated endpoint

### ISSUE THREAD

**Issue by dgroene** [typhonius/acquia-php-sdk-v2#147]
The purge function of Domain is using  
"/environments/${environmentUuid}/domains/actions/clear-varnish"
which is deprecated (http://cloudapi-docs.acquia.com/). Today I noticed that although it states "___ executed varnish cleared" in the event log of the cloud UI, this does not seem to actually clear varnish anymore. When done manually, the report log states "___ executed Clearing caches"
The endpoint for clearing caches is:
/environments/{environmentId}/actions/clear-caches

**Comment by typhonius** [comment]
Hmm the deprecated endpoint should still work (although will be removed in Cloud API v3) and if it's in the Cloud UI it would indicate that the SDK is functioning as expected.

@eporama you might have some extra insight to share here as I don't see an equivalent non-deprecated endpoint but that also may not have been built yet.

**Comment by dgroene** [comment]
I am now unable to reproduce this- varnish clear seems to be working. This may have been something unique to my environment.

**Comment by dgroene** [comment]
Sorry, now I am able to reproduce. This is not a quirk of my environment but a real phenomenon. Did the following:
1.) Loaded a page in stage in incognito window to ensure it is in varnish cache.
2.) Made a minor change to some text in production and copied the database down from prod->stage.
3.) Cleared the Drupal cache in stage using drush
4.) Reloaded the page in incognito window (stage) - observed that my change is not present
5.) Cleared the caches using the clear-varnish endpoint. Event log in Acquia platform showed "___ executed varnish cleared"
6.) Reloaded page in incognito window (stage) - observed that my change is not present
7.) in Acquia cloud platform went to stage environment and manually pressed "clear caches" button. Event log shows "___ executed Clearing Caches"
8.) Reloaded page in incognito window (stage) - observed that my change can NOW be seen

I get that deprecated means the endpoint should still work until V3, but it just isn't the case. "Clear varnish" no longer works to actually clear varnish, and where there used to be a "clear varnish" button in the Cloud UI, it is now "Clear caches" and performs a different operation that actually DOES clear varnish. I think it is necessary to use the clear-caches endpoint (/environments/{environmentId}/actions/clear-caches) in order to make that happen. I will extend the class and change the endpoint, but will leave this here because I think this is a real issue.

**Comment by typhonius** [comment]
No need to extend the class externally, I think a change away from the deprecated endpoint (even as an addition) would be beneficial in this library. 

At a brief look, we might also benefit from adding the environment clear all caches endpoint to the existing clear domain cache endpoint. 

I’ll leave this issue open to track and then whoever gets to the PR first will be able to attach it.

**Comment by eporama** [comment]
I attempted to replicate this and was unable to. I used `curl` instead of a browser, but hit a page, registered a `X-Cache: HIT` and that `X-Cache-Hits` incremented with a couple of attempts.

Then I used `acli api:environments:domain-clear-varnish myapp.prod mydomain.com` and waited about 15 seconds or so for the process to run and then the curl registered `Age: 0` and `X-Cache: MISS` on the same result.

Is there any chance you have Acquia's Platform CDN or Edge or any other CDN in place?
